### PR TITLE
WT-4891 WT_SESSION.verify memory allocation overrun

### PR DESCRIPTION
**Pull request by luke-pearson**

### CODE REVIEW

**Comment by luke-pearson** [comment]
retest this please

File: src/meta/meta_ckpt.c [hunk 1]
```diff
@@ -352,7 +352,7 @@ __wt_meta_ckptlist_get(WT_SESSION_IMPL *session,
 		__wt_config_subinit(session, &ckptconf, &v);
 		for (; __wt_config_next(&ckptconf, &k, &v) == 0; ++slot) {
 			WT_ERR(__wt_realloc_def(
-			    session, &allocated, slot + 1, &ckptbase));
+			    session, &allocated, slot + 2, &ckptbase));
```
Comment (keithbostic) [inline src/meta/meta_ckpt.c]: lgtm, but I think it needs a comment (the other allocation has one), although it has a repeated word, so it's busted.

File: test/csuite/wt4891_meta_ckptlist_get_alloc/main.c [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+main(int argc, char *argv[])
+{
+	TEST_OPTS *opts, _opts;
+	WT_CURSOR *cursor, *cursorp;
+	WT_CURSOR cursor_array[CHECKPOINT_COUNT];
+	WT_SESSION *session2;
+	WT_SESSION *session;
+	opts = &_opts;
+	memset(opts, 0, sizeof(*opts));
+	testutil_check(testutil_parse_opts(argc, argv, opts));
```
Comment (keithbostic) [inline test/csuite/wt4891_meta_ckptlist_get_alloc/main.c]: KNF, the `WT_SESSION` declarations should be on the same line and there should be a blank line after the variable declarations and before the initialization.

File: test/csuite/wt4891_meta_ckptlist_get_alloc/main.c [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+	    session->create(session, opts->uri, "key_format=S,value_format=i"));
+
+	testutil_check(
+	    session->open_cursor(session, opts->uri, NULL, NULL, &cursor));
+
+	/*
+	 * Create checkpoints and keep them active by around by opening a
+	 * checkpoint cursor for each one.
+	 */
+	for (int i = 0; i < CHECKPOINT_COUNT; ++i) {
```
Comment (keithbostic) [inline test/csuite/wt4891_meta_ckptlist_get_alloc/main.c]: KNF, no local variable declarations.

File: test/csuite/wt4891_meta_ckptlist_get_alloc/main.c [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+		testutil_check(session->open_cursor(session, opts->uri, NULL,
+		    "checkpoint=WiredTigerCheckpoint", &cursorp));
+	}
+
+	testutil_check(session->close(session, NULL));
+
+	testutil_check(
+	    opts->conn->open_session(opts->conn, NULL, NULL, &session2));
+
+	testutil_check(session2->verify(session, opts->uri, NULL));
```
Comment (keithbostic) [inline test/csuite/wt4891_meta_ckptlist_get_alloc/main.c]: Trivial simplification: you don't need `session2`, just re-use `session`, you don't need `cursor_array`, the saved `WT_CURSOR` values are never used.

Comment (luke-pearson) [reply test/csuite/wt4891_meta_ckptlist_get_alloc/main.c]: Yeah, I only did those things for readability at a glance, but agreed on all points.

**Pull request merged**